sdks/typescript/client-typescript-core/README.md

```diff
@@ -154,3 +154,96 @@ export enum Readyness {
 }
 ```
 
+## Analytics
+
+Google Analytics works the same way as the other SDKs. When you log an event on the repository,
+it will capture the value of all of the feature flags and featutre values (in case they change),
+and log that event against your analytics provider, once for each feature. This allows you to
+slice and dice your events by state each of the features were in. We send them as a batch, so it
+is only one request.
+
+There are two different implementations, one for when you are in the browser and one for when you
+are in nodejs. You don't need to worry about this, the code detects which one it is in and 
+creates the correct instance. 
+
+In either case, you need to register your implementation with the repository. The only one we
+currently support is Google Analytics, so you need:
+
+- a Google analytics key - usually in the form `UA-123456`. You must provide this up front.
+- a CID - a customer id this is associate with this. You can provide this up front or you can
+provide it with each call, or you can set it later. 
+
+1) You can set it in the constructor:
+
+```typescript
+const collector = new GoogleAnalyticsCollector('UA-123456', 'some-CID');
+```
+
+2) You can tell the collector later.
+
+```typescript
+const collector = new GoogleAnalyticsCollector('UA-123456');
+collector.cid = 'some-value'; // you can set it here
+```
+
+3) When you log an event, you can pass it in the map:
+
+```typescript
+const data = new Map<string, string>();
+data.set('cid', 'some-cid');
+
+featureHubRepository.logAnalyticsEvent('event-name', other: data);
+```
+
+4) For a NODE server, you can set as an environment variable named `GA_CID`.
+
+```typescript
+featureHubRepository.addAnalyticCollector(collector);
+```
+
+As you can see from above (in option 3), to log an event, you simply tell the repository to
+log an analytics event. It will take care of bundling everything up, passing it off to the
+Google Analytics collector which will post it off.
+
+## FeatureHub Test SDK
+
+When doing tests, it is often desirable to update your feature values, particularly flags. We provide an method to do this
+using the `FeatureUpdater` class. Use of the API is based on the rights of your SDK-URL. Generally you should
```

Comment:
I presume before updating a value I can call a feature to find out current state, would it be using the eventsource library from the test?

Reply:
Your repository will already have the value. Its not normal you would do  this however, because in a test you are just wanting to assert the value is true or false, I figured you wouldn't care if it was or wasn't at the moment. Its more like a CEO command.